### PR TITLE
Export more correct types

### PR DESCRIPTION
**Pull request by chriskrycho**
The previous iteration, because it did not test the exported types in any way, had two misses, one serious and one less so:

- the less serious miss was that the object types were incorrectly set up for the tests, and could be simplified as well
- the more serious miss was that the exported type for `TrackedArray` was missing all of the `Array` methods, so downstream consumers could not use it as an array!

### CODE REVIEW

**Comment by chriskrycho** [comment]
NOTE: there's a problem with the updated ESLint config for this, and I haven't yet had a chance to get to the bottom of it. Will try to do so within the next few days.

File: addon/-private/array.ts [hunk 1]
```diff
@@ -3,36 +3,36 @@ import {
   consumeTag,
   dirtyTag,
   consumeCollection,
-  dirtyCollection
-} from 'tracked-maps-and-sets/-private/util';
+  dirtyCollection,
+} from "tracked-maps-and-sets/-private/util";
```
Comment (pzuraq) [inline addon/-private/array.ts:7]: Converting to double quotes seems strange, was this a prettier setting?

Comment (chriskrycho) [reply addon/-private/array.ts:7]: Doh, yes, auto-fix from VS Code. Thought I'd avoided all of these. Will update. Thanks!

**Comment by chriskrycho** [comment]
This should be unblocked by all the dependency work I'm doing across the repo, but I'm going to go ahead and get the dependency situation fully up to date before landing this.

**Comment by chriskrycho** [comment]
Superseded by #56 (which is just this, but rebased and updated).

**Pull request closed without merging**